src/noflo-component-access.js

```diff
@@ -0,0 +1,143 @@
+// noflo-component-access.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = facadeComponent;
+
+/**
+ * Create a noflo.Component facade from a given component. Access to the component
+ * is limited to only select properties and functions. An rpf hash property is
+ * also present for arbitrary assignments and is shared for all facades of the
+ * same Component instance.
+ * This facade mimic the main usage of http://noflojs.org/api/Component.html
+ * Usage:
+ *  var component = componentAccess(new noflo.Component());
+ */
+function facadeComponent(component) {
+    return _.extend({
+        get inPorts() {
+            return _.mapObject(_.pick(component.inPorts, isInPort), facadePort);
+        },
+        get outPorts() {
+            return _.mapObject(_.pick(component.outPorts, isOutPort), facadePort);
+        },
+        get rpf() {
+            return component.rpf = component.rpf || {};
+        }
+    }, facadeFunctions(component,
+        'getDescription',
+        'getIcon'
+    ), facadeEventEmitter(component));
+}
+
+/**
+ * Create a noflo Port facade from a given noflo Port that includes only a limited
+ * subset of properties and functions.
+ * This facade mimic the main usage of http://noflojs.org/api/OutPort.html
+ */
+function facadePort(port) {
+    return _.extend({
+        // port.nodeInstance might not be assigned yet
+        get nodeInstance() {
+            return port.nodeInstance && facadeComponent(port.nodeInstance)
```

Comment:
Won't this be creating a new facadeComponent every time port.nodeInstance is accessed at runtime?  I think the facade should only be created once for each component instance.